File: quickSort.cpp
#include<bits/stdc++.h>
using namespace std;
inline vector<long long>add(vector<long long>leftArray,long long pivot,vector<long long>rightArray){
    vector<long long>ans;
    leftArray.push_back(pivot);
    for(auto&it:leftArray)ans.push_back(it);
    for(auto&it:rightArray)ans.push_back(it);
    return ans;
}
vector<long long> quickSort(vector<long long>&array){
    if(array.size()<2)return array;
    vector<long long> leftArray,rightArray;
    long long pivot=array[array.size()-1];
    for(long long i=0;i<array.size()-1;i++){
        if(array[i]<pivot){
            leftArray.push_back(array[i] );
        }else{
            rightArray.push_back(array[i]);
        }
    }
    return add(quickSort(leftArray),pivot,quickSort(rightArray));
}
int main(){
    vector< long long> array={1,2,3,4,5,6,7,3,2,2,4,4,5,32,2,23,4,423,4,8,4,0};
    for(auto&it: quickSort(array))cout<<it<<' ';
    // quickSort()
}

File: QuickSort.js
function quickSort(array, ascending = true) {
  if (array.length < 2) return array;
  let pivot = array[array.length - 1];
  let leftArray = [];
  let rightArray = [];
  for (let i = 0; i < array.length - 1; i++) {
    if (array[i] > pivot) {
      if (ascending) rightArray.push(array[i]);
      else leftArray.push(array[i]);
    } else {
      if (ascending) leftArray.push(array[i]);
      else rightArray.push(array[i]);
    }
  }
  return [
    ...quickSort(leftArray, ascending),
    pivot,
    ...quickSort(rightArray, ascending),
  ];
}

let array = [
  1, 2, 3, 34, 5, 56, 7, 4, 0, 343, 2, 21, 1, 1, 23, 34, 9, -3, 986, 7, 53,
];
console.log(quickSort(array));
console.log(quickSort(array, false));
